kategory/src/test/kotlin/kategory/free/CoYonedaTest.kt

```diff
@@ -1,17 +1,39 @@
 package kategory
 
 import io.kotlintest.KTestJUnitRunner
+import io.kotlintest.matchers.shouldBe
 import io.kotlintest.properties.forAll
 import org.junit.runner.RunWith
 
 @RunWith(KTestJUnitRunner::class)
 class CoYonedaTest : UnitSpec() {
+    val F = CoYoneda.functor<Id.F, Int>()
+
+    val EQ = object : Eq<CoYonedaKind<Id.F, Int, Int>> {
+        override fun eqv(a: CoYonedaKind<Id.F, Int, Int>, b: CoYonedaKind<Id.F, Int, Int>): Boolean =
+                a.ev().lower(Id) == a.ev().lower(Id)
+
+    }
+
+    val AP: Applicative<CoYonedaF<Id.F, Int>> = object : Applicative<CoYonedaF<Id.F, Int>> {
+        override fun <A> pure(a: A): CoYonedaKind<Id.F, Int, A> =
+                CoYoneda.apply(Id(0), { a })
+
+        override fun <A, B> map(fa: CoYonedaKind<Id.F, Int, A>, f: (A) -> B): CoYonedaKind<Id.F, Int, B> =
+               F.map(fa, f)
+
+        override fun <A, B> ap(fa: CoYonedaKind<Id.F, Int, A>, ff: CoYonedaKind<Id.F, Int, (A) -> B>): CoYonedaKind<Id.F, Int, B> =
```

Comment:
Instead of implementing an impossible instance here we should make the generator parametric and not require `Applicative`. Instead of `genApplicative` we can require something that provides `(A) -> HK<F, A>` which to most instances it would be `Applicative#pure`.
In cases like this one it would just be `a -> CoYoneda.apply(Id(0), { a })` so there is no need to be forced to implement `Applicative#ap`.